sda-commons-server-spring-data-mongo/src/main/java/org/sdase/commons/server/spring/data/mongo/MongoConfiguration.java

```diff
@@ -216,22 +23,12 @@ public MongoConfiguration setConnectionString(String connectionString) {
   }
 
   /**
-   * We either need the
-   *
-   * <ul>
-   *   <li>'connectionString'
-   *   <li>'hosts' and 'database'
-   * </ul>
-   *
-   * @return true if the configuration is valid
+   * @return true if the connectionString is not blank or empty
    */
   @AssertTrue
   @JsonIgnore
   public boolean isValid() {
-    if (StringUtils.isNotBlank(connectionString)) {
-      return true;
-    }
 
-    return StringUtils.isNotBlank(hosts) && StringUtils.isNotBlank(database);
+    return StringUtils.isNotBlank(connectionString);
```

Comment:
Imho we can use `@NotBlank` at the property now and remove this extra method. It was just introduced to support _„either old configuration or connection string must not be blank“_ without a custom validator implementation.

Reply:
[The test](https://github.com/SDA-SE/sda-dropwizard-commons/pull/2949/files#diff-0f5115726af4a33bb13c596c946fb164da0c29a67a17801b07e7bb447642ccacR10) could then use a Validator.